src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,314 @@
+//
+//  GeolocationPlugin.m
+//  Geolocation
+//
+//  Created by dev on 2/9/16.
+//
+//
+
+#import "GeolocationPlugin.h"
+
+@implementation GeolocationPlugin
+
+-(void)start:(CDVInvokedUrlCommand *)command
+{
+    // NSString *callbackId = command.callbackId;
+    //  [self returnLocationInfo:callbackId andKeepCallback:"OKay"];
+    
+    CDVPluginResult* pluginResult = nil;
+    coord = [command.arguments objectAtIndex:0];
+    
+    
+    //[self InitData];
+    
+    if (![self.thread isExecuting]) {
+        NSLog(@"Starting thread...");
+        loopflag = TRUE;
+        [self.thread start];
+    }
+}
+
+-(NSThread*)thread{
+    if (!_thread) {
+        _thread = [[NSThread alloc]initWithTarget:self selector:@selector(longloop) object:nil];
```

Comment:
since we're performing location updates and region monitoring, afaik starting this once is enough, however it has to be [restarted when the app is put in foreground](http://stackoverflow.com/a/23127411/890242). hence the thread does not need to loop IMO.

also to keep getting location updates in background mode, see [this guide](https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/LocationAwarenessPG/CoreLocation/CoreLocation.html) => search for "keep monitoring". 

http://stackoverflow.com/a/25939298/890242